contracts/optimism/OpStackTokenRateObserver.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {ITokenRateObserver} from "../lido/interfaces/ITokenRateObserver.sol";
+import {L1LidoTokensBridge} from "./L1LidoTokensBridge.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+/// @author kovalgek
+/// @notice Pushes token rate when rebase event happens.
+contract OpStackTokenRateObserver is ERC165, ITokenRateObserver {
```

Comment:
Is this specific for OpStack?
I expect non-opstack bridges very likely to have the same interface `pushTokenRate(L2_GAS_LIMIT_FOR_PUSHING_TOKEN_RATE)`

Thus it could be just `GeneralTokenRateObserver`.

Reply:
As we agreed during the call, we can't create a general solution for the observer; thus, it isn't valid anymore.